include/fiction/technology/charge_distribution_surface.hpp

```diff
@@ -316,6 +316,26 @@ class charge_distribution_surface<Lyt, false> : public Lyt
 
         this->charge_distribution_to_index();
     }
+    /**
+     * This function can be used to detect which SiDBs must be negatively charged due to their location. Important:
+     * This function must be applied to a charge layout where all SiDBs are negatively initialized.
+     *
+     * @return Vector of indices describing which SiDBs must be negatively charged.
+     */
+    std::vector<uint64_t> negative_sidb_detection() noexcept
+    {
+        std::vector<uint64_t> negative_sidbs{};
+        negative_sidbs.reserve(this->num_cells());
+        this->foreach_cell(
+            [&negative_sidbs, this](const auto& c)
+            {
+                if (-*this->get_local_potential(c) + strg->phys_params.mu < -physical_constants::POP_STABILITY_ERR)
```

Comment:
Shouldn't you first check whether local potential `has_value()`?

Reply:
good point!